src/store/index.js

```diff
@@ -63,6 +75,32 @@ export default new Vuex.Store({
         console.log(err);
       }
     },
+    async login({ commit, getters, dispatch }, credentials) {
+      if (getters.isLoggedIn) await dispatch("logout");
+      const response = await authApi.login(credentials);
+      const authHeader = response.headers.authorization;
+      setAuthHeader(authHeader);
+      commit("setUserData", { authHeader });
+    },
+    async logout({ commit, getters }) {
+      if (!getters.isLoggedIn) return;
+      // TODO: API action
```

Comment:
we already have this action so maybe it's worth to add it 🙈

Reply:
Done 🙈